src/main/java/com/netgrif/application/engine/workflow/service/WorkflowService.java

```diff
@@ -172,6 +178,14 @@ public Page<Case> getAll(Pageable pageable) {
         return setImmediateDataFields(page);
     }
 
+    @Override
+    public Page<Case> findAllByUri(String uri, Pageable pageable) {
+        Page<Case> page = repository.findAllByUriNodeId(uri, pageable);
+        page.getContent().forEach(this::setPetriNet);
+        decryptDataSets(page.getContent());
```

Comment:
why does findAllByUri decrypt the dataset, but the search implementation bellow doesn't? The other search implementation does. Maybe add a private/protected method that converts a collection of cases to a response and use it in all controller methods?